### PR TITLE
DC: Fixes crash when no validation_set is passed (auto)

### LINKED ISSUE #2691: Drawing classifier crashes on master with no validation set option
Following the user guide dataset, here is the error I get on drawing classifier
```
import turicreate as tc

# Try any one of the following
SFRAME_PATH = "sframes/bitmap_square_triangle.sframe"

# Load the data
data =  tc.SFrame(SFRAME_PATH)

# Make a small train-test split since our toolkit is not very data-hungry 
# for 2 classes
train_data, test_data = data.random_split(0.7)

# Create a model
model = tc.drawing_classifier.create(train_data, 'label')
```

```
non-resource variables are not supported in the long term
2019-11-19 18:37:44.716262: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-11-19 18:37:44.744622: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x7fecfae2e650 executing computations on platform Host. Devices:
2019-11-19 18:37:44.744651: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Host, Default Version
Using CPU to create model
+--------------+----------------+---------------------+--------------+
| Iteration    | Train Accuracy | Validation Accuracy | Elapsed Time |
+--------------+----------------+---------------------+--------------+
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
~/miniconda3/envs/tc-6.0/lib/python3.6/site-packages/turicreate/extensions.py in __run_class_function(self, fnname, args, kwargs)
    273         try:
--> 274             ret = self._tkclass.call_function(fnname, argument_dict)
    275         except RuntimeError as exc:

cy_model.pyx in turicreate._cython.cy_model.UnityModel.call_function()

cy_model.pyx in turicreate._cython.cy_model.UnityModel.call_function()

RuntimeError: An error occurred: IndexError: arrays used as indices must be of integer (or boolean) type
```

### PR DESCRIPTION
**Pull request by shreyajain17**
fixes #2691

### CODE REVIEW

File: src/python/turicreate/toolkits/drawing_classifier/_tf_drawing_classifier.py [hunk 1]
```diff
@@ -182,7 +182,8 @@ def predict(self, feed_dict):
         if is_train:
             # convert to one hot
             labels = feed_dict["labels"].astype("int32").T
-            one_hot_labels[_np.arange(num_samples), labels] = 1
+            one_hot_labels[_np.arange(int(num_samples)), labels] = 1
```
Comment (srikris) [inline src/python/turicreate/toolkits/drawing_classifier/_tf_drawing_classifier.py]: Why is num_samples not an int in the first place?

Comment (shreyajain17) [reply src/python/turicreate/toolkits/drawing_classifier/_tf_drawing_classifier.py]: Well when you send it from C++, we send it as `shared_float_array` which is float32.

Comment (srikris) [reply src/python/turicreate/toolkits/drawing_classifier/_tf_drawing_classifier.py]: Can you add a simple unit test to make sure we don't crash again.

Comment (shantanuchhabra) [reply src/python/turicreate/toolkits/drawing_classifier/_tf_drawing_classifier.py]: Added.

**Pull request merged**